server/field/lifepool/pool.go

```diff
@@ -263,22 +263,50 @@ func (pool *Data) NpcAcknowledge(poolID int32, plr controller, data []byte) {
 // MobAcknowledge bytes to be applied to the pool
 func (pool *Data) MobAcknowledge(poolID int32, plr controller, moveID int16, skillPossible bool, action byte, skillData uint32, moveData movement.Data, finalData movement.Frag, moveBytes []byte) {
 	for i, v := range pool.mobs {
+		mob := pool.mobs[i]
+
 		if poolID == v.SpawnID() && v.Controller().Conn() == plr.Conn() {
 			skillID := byte(skillData)
 			skillLevel := byte(skillData >> 8)
 			skillDelay := int16(skillData >> 16)
 
-			actualAction := int(byte(action >> 1))
+			var actualAction int8
 
 			if action < 0 {
 				actualAction = -1
+			} else {
+				actualAction = int8(action) >> 1
 			}
 
+			// Perform either skill or attack
 			if actualAction >= 21 && actualAction <= 25 {
 				pool.mobs[i].PerformSkill(skillDelay, skillLevel, skillID)
 			} else if actualAction > 12 && actualAction < 20 {
-				pool.mobs[i].PerformAttack(byte(actualAction - 12))
+				attackID := byte(actualAction - 12)
+
+				// check mob can use attack
+				if level, valid := mob.Skills[attackID]; valid {
+					levels, err := nx.GetMobSkill(attackID)
+
+					if err != nil {
+						return
+					}
+
+					if int(level) < len(levels) {
+						skill := levels[level]
+						mob.SetMP(mob.MP() - skill.MpCon)
+						if mob.MP() < 0 {
+							mob.SetMP(0)
+						}
+					}
+
+				}
+
+				pool.mobs[i].PerformAttack(attackID)
 			}
+			mob.SetSkillPossible(skillPossible)
```

Comment:
I'm not sure if this is the correct behaviour. The way I see it is as follows:
1. Controller sends a request to use a skill (skillPossible)
2. Server then selects an off-cooldown skill
3a. If a skill is possible then skillPossible is set to true and a skillID and level is sent
3b. If a skills is not possible then skillPossible is set to false and skillID and level are set to zero

I think in the 3b scenario the bool will be set to true in the packet as skillPossible is being passed through but not changed? I think `CanUseSkill` should take a boolean and return bool, byte, byte where the boolean is either true or false depending on if a skill is selected.

Note: I am not sure if my initial assumption in point 1 is correct.

Reply:
I agree with this. This was the method I was initially envisioning as well, and I will refactor the function to act accordingly.

Reply:
I don't think is a need to set skillPossible at all actually. I think this variable is only needed within the scope of choosing a skill. Otherwise the client should let us know when a skill is possible again. It seems to be working fine in new implementation.

Reply:
I have checked out the PR locally and mobs appear to be acting properly. Will you be setting the skill cooldowns to the values defined in nx? Will you add the mob mp regen in this PR?

Reply:
Yes cooldowns will be based on the values in nx file. Have not looked much into mp regen, but I can try to implement that as well this PR.